### ISSUE TITLE
Train localization cues for different datasets

### ISSUE THREAD

**Issue by nightrome** [kolesman/SEC#8]
Hi,
I was wondering whether you could make the files available to train the localization cues on different datasets. Specifically I assume that to create the [deploy.prototxt](https://github.com/kolesman/SEC/blob/master/weak-localization/background/deploy.prototxt) you:
- Removed Dropout
- Removed the annotations
- Removed the loss
- And the solver.prototxt was identical to [this one](https://github.com/kolesman/SEC/blob/master/training/solver.prototxt)

It would also be great if you could share the script to create the .pickle files.

**Comment by kolesman** [comment]
Hi Holger,

Yes, you are right. In particular, I've removed the dropout after the layer 'fc6' and 'fc7' with dropping rate of 0.5.

The 'solver.prototxt' is identical to the you one you've linked, except for one parameter: the "stepsize" is equal to 4000.

Currently I do not have easy-to-run script to generate localization cues. It will require some polishing and adapting my old code to the new naming conventions of this repo.

**Comment by sun11** [comment]
The loss function and input image resize or crop is the most important. Could you explain what is "multilabel logistic loss" in detail? AFAIK, there is a squared loss which used in https://arxiv.org/pdf/1406.5726.pdf.

**Comment by yaoqi-zd** [comment]
@kolesman Hi, do you mean, to train the foreground cues, I only need to change the 'stepsize' in solver.prototxt, and add this  [deploy.prototxt](https://github.com/kolesman/SEC/blob/master/weak-localization/foreground/deploy.prototxt) with:
1. 'ImageData' layer to provide images;
2. annotation layer(like in pylayers) to provide image labels; 
3. 'SigmoidCrossEntropy' layer to compute the multilabel logistic loss.
Since there already exist dropout after 'fc6' and 'fc7' in this deploy.prototxt ?

**Comment by yaoqi-zd** [comment]
@nightrome Hi Holger, have you tried to reproduce the initial forground cues by yourself? I have treid to train the classification network by myself but my generated forground cues are not as good as the provided

**Comment by nightrome** [comment]
@yaoqi-zd No sorry, I haven't pursued this any further.

**Comment by danea32** [comment]
> @nightrome Hi Holger, have you tried to reproduce the initial forground cues by yourself? I have treid to train the classification network by myself but my generated forground cues are not as good as the provided

could you please  share me  the script to create the .pickle files, i want to train it in my own dataset, thank you very much~

**Comment by zmll** [comment]
> @nightrome Hi Holger, have you tried to reproduce the initial forground cues by yourself? I have treid to train the classification network by myself but my generated forground cues are not as good as the provided

could you please share me the script to create the .pickle files, i want to train it , thank you very much!

**Comment by zmll** [comment]
@yaoqi-zd could you please share me the script to create the .pickle files, i want to train the model,but i failed. thank you very much~

**Comment by yaoqi-zd** [comment]
> @yaoqi-zd could you please share me the script to create the .pickle files, i want to train the model,but i failed. thank you very much~

Hi, you can refer to the function 'get_localization_cues_sec' in my code which generates the seeds with cam and external saliency maps.

**Comment by zmll** [comment]
> > @yaoqi-zd could you please share me the script to create the .pickle files, i want to train the model,but i failed. thank you very much~
> 
> Hi, you can refer to the function 'get_localization_cues_sec' in my code which generates the seeds with cam and external saliency maps.

Ok, I will try it,thank you
